### ISSUE TITLE
`rubocop` is easily confused by `%{%{0}}`

### ISSUE THREAD

**Issue by postmodern** [rubocop/rubocop#11618]
I stumbled across a weird edge-case where a `%{ }` string containing PowerShell (don't ask) also contained `%{0}`. Ruby does allow for %{} to exist within `%{ ... }` strings and will not interpret it differently. However, `rubocop` get's confused by `%{}` existing within a `%{ ... }` string, mainly the `Style/PercentLiteralDelimiters` and `Style/FormatStringToken` rules.

## Expected behavior

No warnings.

## Actual behavior

```
test.rb:3:6: C: [Correctable] Style/PercentLiteralDelimiters: %-literals should be delimited by ( and ).
puts %{%{0}}
     ^^^^^^^
test.rb:3:8: C: [Correctable] Style/FormatStringToken: Prefer annotated tokens (like %<foo>s) over template tokens (like %{foo}).
puts %{%{0}}
       ^^^^
```

## Steps to reproduce the problem

`Gemfile`:

```ruby
source 'https://rubygems.org'

gem 'rubocop'
```

`test.rb`:

```ruby
# frozen_string_literal: true

puts %{%{0}}
```

1. `bundle install`
2. `bundle exec rubocop test.rb`

## RuboCop version

```
$ bundle exec rubocop -V
1.45.1 (using Parser 3.2.1.0, rubocop-ast 1.26.0, running on ruby 3.1.3) [x86_64-linux]
```

**Comment by jonas054** [comment]
First of all, I think `Style/PercentLiteralDelimiters` behaves exactly as expected here. Round parentheses are the default, so the outer `%{...}` is reported as an offense.

When it comes to `Style/FormatStringToken`, I gather from its documentation that for the default style, it doesn't bother to distinguish between typical formatting context and other context. This is why the `puts` line is reported even though we can see that the inner `%{0}` is clearly not a format expression.

So it's not necessarily a bug. It's more the result of a design decision to report everything that looks like a format string.

**Comment by dvandersluis** [comment]
I have opened #13974 to update the documentation of `Style/FormatStringToken`.
